epics/pv.py

```diff
@@ -295,17 +295,15 @@ def get(self, count=None, as_string=False, as_numpy=True,
 
         if count is None:
             count = len(val)
-        if (as_numpy and ca.HAS_NUMPY and 
-            not isinstance(val, ca.numpy.ndarray)):
-            if count == 1:
-                val = [val]
-            val = ca.numpy.array(val)
-        elif (as_numpy and ca.HAS_NUMPY and count == 1 and
-            not isinstance(val, ca.numpy.ndarray)):
-            val = ca.numpy.array([val])
-        elif (not as_numpy and ca.HAS_NUMPY and
-              isinstance(val, ca.numpy.ndarray)):
-            val = list(val)
+
+        if ca.HAS_NUMPY:
+            is_array = isinstance(val, ca.numpy.ndarray)
+            if as_numpy and not is_array:
+                if not isinstance(val, list):
+                    val = [val]
+                val = ca.numpy.array(val)
+            elif not as_numpy and is_array:
+                val = val.tolist()
```

Comment:
Is this change meant to be a cleanup for clarity?  Because the behavior is actually different.